call_apply_bind_exercise/callApplyBind.js

```diff
@@ -0,0 +1,93 @@
+///////////////////////////////////////////////////////////
+// 1: fix the following code
+///////////////////////////////////////////////////////////
+
+var obj = {
+    fullName: "Harry Potter",
+    person: {
+        sayHi: function(){
+            return "This person's name is " + obj.fullName;
+        }
+    }
+}
+
+// call the function: 
+
+obj.person.sayHi();
+
+///////////////////////////////////////////////////////////
+// 2: examples of "array-like objects"
+///////////////////////////////////////////////////////////
+
+// 1 - arguments - a keyword created for each function; arguments.length works but no other array functions can be called on arguments because it's not an actual array
+
+// 2 - ???
+
+///////////////////////////////////////////////////////////
+// 3. sumEvenArguments
+///////////////////////////////////////////////////////////
+
+function sumEvenArguments(...args) {
+	var sum = 0;
+	for (let val of args) {
+		if (val % 2 === 0) {
+			sum += val;
+		}
+	}
+	return sum
+}
+
+
+///////////////////////////////////////////////////////////
+// 4. arrayFrom
+///////////////////////////////////////////////////////////
+
+
+function arrayFrom(arrLikeObj) {
+	return [].concat.apply(this, arrLikeObj);
```

Comment:
You don't need concat, you can do this using slice - use concat for nested arrays.